extensions/logs/suite.go

```diff
@@ -0,0 +1,214 @@
+}
+
+type Suite struct {
+	suite.Suite
+
+	kubeClient kubernetes.Interface
+
+	testStartTime time.Time
+	nsmContainers map[types.UID]bool
+	logQueue      chan logItem
+	waitGroup     sync.WaitGroup
+}
+
+type logItem struct {
+	namespace  string
+	pod        string
+	logDir     string
+	logOptions *corev1.PodLogOptions
+}
+
+func init() {
+	if err := envconfig.Usage("", &config); err != nil {
+		panic(err)
+	}
+
+	if err := envconfig.Process("", &config); err != nil {
+		panic(err)
+	}
+}
+
+func (s *Suite) SetupSuite() {
+	var err error
+
+	s.kubeClient, err = newKubeClient()
+	require.NoError(s.T(), err)
+
+	s.logQueue = make(chan logItem)
+	for i := 0; i < config.LogWorkerCount; i++ {
+		s.waitGroup.Add(1)
+		go func() {
+			defer s.waitGroup.Done()
+			for src := range s.logQueue {
+				s.saveLog(src)
+			}
+		}()
+	}
+}
+
+func (s *Suite) TearDownSuite() {
+	close(s.logQueue)
+	s.waitGroup.Wait()
+}
+
+func (s *Suite) SetupTest() {
+	s.testStartTime = time.Now()
+	s.nsmContainers = make(map[types.UID]bool)
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.ContextTimeout)
+	defer cancel()
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	require.NoError(s.T(), err)
+
+	for podIdx := range pods.Items {
+		pod := &pods.Items[podIdx]
+
+		if !isExcludedNamespace(pod.Namespace) {
+			s.nsmContainers[pod.UID] = true
+		}
+	}
+}
+
+func (s *Suite) AfterTest(suiteName, testName string) {
+	logDir := filepath.Join(config.ArtifactsDir, suiteName, testName)
+	require.NoError(s.T(), os.MkdirAll(logDir, os.ModePerm))
+
+	logOptions := corev1.PodLogOptions{
+		Timestamps: true,
+		SinceTime:  &metav1.Time{Time: s.testStartTime},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.ContextTimeout)
+	defer cancel()
+
+	pods, err := s.kubeClient.CoreV1().Pods("").List(ctx, metav1.ListOptions{})
+	require.NoError(s.T(), err)
+
+	var waitGroup sync.WaitGroup
+	for podIdx := range pods.Items {
+		pod := &pods.Items[podIdx]
+
+		if isExcludedNamespace(pod.Namespace) {
+			continue
+		}
+
+		nextLogItem := logItem{
+			namespace:  pod.Namespace,
+			pod:        pod.Name,
+			logDir:     logDir,
+			logOptions: &logOptions,
+		}
+
+		if _, ok := s.nsmContainers[pod.UID]; ok {
+			s.logQueue <- nextLogItem
+			continue
+		}
+
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+			s.saveLog(nextLogItem)
+		}()
+	}
+
+	waitGroup.Wait()
+}
+
+func (s *Suite) saveLog(src logItem) {
+	ctx, cancel := context.WithTimeout(context.Background(), config.ContextTimeout)
+	defer cancel()
+
+	data, err := s.kubeClient.CoreV1().
+		Pods(src.namespace).
+		GetLogs(src.pod, src.logOptions).
+		DoRaw(ctx)
+
+	require.NoError(s.T(), err)
+
+	if len(data) > 0 {
+		logFile := filepath.Join(src.logDir, src.pod+".log")
+		require.NoError(s.T(), ioutil.WriteFile(logFile, data, os.ModePerm))
+	}
+}
+
+func envOrDefault(key, defaultValue string) string {
```

Comment:
This function actually is not needed. Please add KUBECONFIG into envconfig with default value.